changelog.d/20240702_175644_30907815+rjmello_exec_cmd_sc_34376.rst

```diff
@@ -0,0 +1,11 @@
+New Functionality
+^^^^^^^^^^^^^^^^^
+
+- Added the ``globus-compute-endpoint python-exec`` command to run Python modules as scripts
```

Comment:
This isn't exactly a user-facing change. It might be better to put this under Changed as `we no longer rely on PATH to resolve ... instead we use a new mechanism ... for more reliable worker starts for multi user endpoint users`. @rjmello what do you think?

Reply:
I intentionally split the work into two separate changelog fragments: one for the addition of the the new `python-exec` CLI command, which I placed under New Functionality, and another for utilizing it to launch workers, which I placed under Changed. Although we currently only use the `python-exec` command to launch workers, I expect that we will use it for other purposes within Globus Compute (e.g., launching the Parsl interchange) or that dependent projects will find it useful.

Reply:
I intentionally split the work into two separate changelog fragments: one for the addition of the the new python-exec CLI command, which I placed under New Functionality, and another for utilizing it to launch workers, which I placed under Changed. Although we currently only use the python-exec command to launch workers, I expect that we will use it for other purposes within Globus Compute (e.g., launching the Parsl interchange) or that dependent projects will find it useful.